### PR TITLE
Gardening: Migrate CHANGELOG.md to GitHub issues

### PR DESCRIPTION
**Pull request by AnthonyLatsis**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -879,7 +879,7 @@ Swift 5.5
   The `self` parameter of actor methods are implicitly `isolated`. The
   `nonisolated` keyword makes the `self` parameter no longer `isolated`.
 
-* [SR-14731][]:
+* [#57081][]:
```
Comment (xedin) [inline CHANGELOG.md:882]: Should we mention that this is an `Issue` explicitly here for readability?

Comment (AnthonyLatsis) [reply CHANGELOG.md:882]: Do you believe this distinction is worth the additional effort of maintaining consistency? All the more if we start linking pull requests like `Pull #[0-9]+` for entries that were not tracked by an issue. Another approach that feels less prone to format variations is to use the URL path as the link text: `[apple/issues/12345][]:` or `[apple/pull/12345][]:`. 

@hborla A third opinion is welcome!

Comment (xedin) [reply CHANGELOG.md:882]: I like the URL path idea!

Comment (benrimmington) [reply CHANGELOG.md:882]: SwiftPM's [CHANGELOG.md][] links to pull requests with the bare `#1234` format.

[CHANGELOG.md]: <https://github.com/apple/swift-package-manager/blob/main/CHANGELOG.md>

Comment (xedin) [reply CHANGELOG.md:882]: Hm, we probably don’t want to reference pulls in the CHANGELOG so maybe having#<issue> in there is going to be self-evident…

**Comment by AnthonyLatsis** [comment]
@hborla ping

File: CHANGELOG.md (hunk 1 above)
Comment (hborla) [reply CHANGELOG.md:882]: I don't have a strong opinion either way. I think what's important in the CHANGELOG is the description of the user-facing change. IMO the plain `#` isn't confusing, and when viewing the file it's clearly a link that you can click on for more information

**Review by hborla — APPROVED** [review]
Thank you! (and thanks for the ping 🙂)

**Comment by AnthonyLatsis** [comment]
@swift-ci please smoke test

File: CHANGELOG.md (hunk 1 above)
Comment (AnthonyLatsis) [reply CHANGELOG.md:882]: > Hm, we probably don’t want to reference pulls in the CHANGELOG so maybe having # in there is going to be self-evident

Issues are obviously the more appropriate place for developers and compiler engineers to discuss or provide feedback on bugs/improvements/etc., plus issues can take multiple incremental changes in multiple repos to close, so I believe it is a sensible convention to not reference pulls directly, and one that is likely to hold.

**Pull request merged**